app/models/user.rb

```diff
@@ -86,6 +86,46 @@ class User < ActiveRecord::Base
     },
     ranked_by: ":tsearch + (0.25 * :trigram)"
 
+  def self.subset_selection(ending_in = 5)
+    subset_selection = "id%10 = ?"
+    where(subset_selection, ending_in)
+  end
+
+  def self.dormant(window=14, query=User)
+    DatabaseReplica.read("read_dormant_users_from_replica") do
+      # devise trackable sets the current_sign_in_at on each login
+      havent_signed_in_since = "date(now()) - date(current_sign_in_at) >= #{window}"
+      query = query.where(valid_email: true).where(havent_signed_in_since)
+      query = query.where(activated_state: User.activated_states[:active]).select(:id)
```

Comment:
I assume we should also be removing users that have `global_email_communication` set to false here?

Reply:
I suppose not unless we're saying a dormant user email is not a general zooniverse email update? Probably best not to start emailing people who have chosen to opt out of general emails.